script_to_db.py

```diff
@@ -1,74 +1,99 @@
+def fetch_data(page_url, page_auth):
+    '''Function to get response from ona api in json format'''
+    response = requests.get(url=page_url, auth=page_auth, timeout=10)
+    return response.json()
 
-def fetch_data():
-    response=requests.get(url=URL,auth=ona_auth)
-    ona_database_file=response.json()
-    return ona_database_file
- 
 
 def create_table():
-    commands=('CREATE TABLE IF NOT EXISTS test_2(_id int primary key NOT NULL,_submitted_by varchar NOT NULL,specific_time time);')
-    conn=None
+    '''Command to create a table in the postgres database'''
+    commands = '''CREATE TABLE IF NOT EXISTS test_2(
+        _id int primary key NOT NULL,
+        _submitted_by varchar NOT NULL,
+        specific_time time);'''
+    execute_statements(commands)
+
+
+def insert_statements():
+    '''Generating insert statement to populate table in postgres'''
+    tupl_lst = [tuple(x) for x in table_values]
+    for lst in tupl_lst:
+        insert_command = f'{"insert into test_2(_id,_submitted_by,specific_time)values(%s,%s,%s);"}'
+        execute_statements(insert_command, *lst)
+
+
+def connection_to_db():
+    '''Function to connect to the postgres database'''
     try:
-        connection=psycopg2.connect(user=USER,password=PASSWORD,host=HOST,port=PORT,database=DATABASE)
-        conn=connection.cursor()
-        
-        if connection:
-            print('success')
-        conn.execute(commands)
-        connection.commit()
-        connection.close()
-    except Exception as error:
+        connection = psycopg2.connect(
+            user=USER,
+            password=PASSWORD,
+            host=HOST,
+            port=PORT,
+            database=DATABASE
+            )
+    except ValueError as error:
         print(error)
-  
-
-def evaluate_json_values():
-    a=fetch_data()
-    var_list=[]
-    for entry in a:
-        first_value=entry["_id"]
-        second_value=entry["_submitted_by"]
-        third_value=entry.get("specific_time")
-        values_to_insert=[(first_value),(second_value),(third_value)]
-        var_list.append(values_to_insert)
-
-    return var_list
+    return connection
 
 
-def insert_statement(table_values):
-    tupl_lst= [tuple (x) for x in table_values]
-    conn=None
+def execute_statements(sql_command, *placeholder_values):
+    '''Function to execute commands'''
     try:
-        connection=psycopg2.connect(user=USER,password=PASSWORD,host=HOST,port=PORT,database=DATABASE)
-        conn=connection.cursor()
-        if connection:
-            print('success')
-        for x in tupl_lst:
-           insert_command=f'insert into test_2(_id,_submitted_by,specific_time) values(%s,%s,%s)'
-           conn.execute(insert_command,x)
-        connection.commit()
-        connection.close()
-    except Exception as error:
+        connection = connection_to_db()
+        conn = connection.cursor()
+        if placeholder_values is None:
+            conn.execute(sql_command)
+            connection.commit()
```

Comment:
This can be simplified by having connection.commit() outside of the if-else statement. No need to have it in both L66 and L69